spacy_llm/tasks/entity_linking.py

```diff
@@ -0,0 +1,310 @@
+
+    @property
+    def _cfg_keys(self) -> List[str]:
+        return ["_template"]
+
+    @property
+    def _Example(self) -> Type[EntityLinkingExample]:
+        return EntityLinkingExample
+
+    def _create_prompt_example(
+        self, example: Example
+    ) -> Optional[EntityLinkingExample]:
+        """Create a entity linking prompt example from a spaCy example."""
+        # Ensure that all entities have their knowledge base IDs set.
+        n_ents = len(example.reference.ents)
+        n_set_kb_ids = sum([ent.kb_id != 0 for ent in example.reference.ents])
+        if n_ents and n_ents != n_set_kb_ids:
+            warnings.warn(
+                f"Not all entities in this document have their knowledge base IDs set ({n_set_kb_ids} out of "
+                f"{n_ents}). Ignoring example:\n{example.reference}"
+            )
+            return None
+
+        # Assemble example.
+        mentions = [ent.text for ent in example.reference.ents]
+        # Fetch candidates. If true entity not among candidates: fetch description separately and add manually.
+        cands_ents_info, solutions = self._fetch_entity_info(example.reference)
+        # If we are to use available docs as examples, they have to have KB IDs set and hence available solutions.
+        assert all([sol is not None for sol in solutions])
+
+        return EntityLinkingExample(
+            text=EntityLinkingTask._highlight_ents_in_text(example.reference),
+            mentions_str=", ".join([f"*{mention}*" for mention in mentions]),
+            mentions=mentions,
+            entity_descriptions=[
+                cands_ent_info[1] for cands_ent_info in cands_ents_info
+            ],
+            entity_ids=[cands_ent_info[0] for cands_ent_info in cands_ents_info],
+            solutions=solutions,
+            reasons=[""] * len(mentions),
+        )
+
+    @staticmethod
+    def _highlight_ents_in_text(doc: Doc) -> str:
+        """Highlights entities in doc text with **.
+        doc (Doc): Doc whose entities are to be highlighted.
+        RETURNS (str): Doc text with highlighted entities.
+        """
+        text = doc.text
+        for i, ent in enumerate(doc.ents):
+            text = (
+                text[: ent.start_char + i * 2]
+                + f"*{ent.text}*"
+                + text[ent.end_char + i * 2 :]
+            )
+
+        return text
+
+    def _fetch_entity_info(
+        self, doc: Doc
+    ) -> Tuple[List[Tuple[List[str], List[str]]], List[Optional[str]]]:
+        """Fetches entity IDs & descriptions and determines solution numbers for entities in doc.
+        doc (Doc): Doc to fetch entity descriptions and solution numbers for. If entities' KB IDs are not set,
+            corresponding solution number will be None.
+        RETURNS (Tuple[List[Tuple[List[str], List[str]]], List[str]]): For each mention in doc: list of candidates
+            with ID and description, list of correct entity IDs.
```

Comment:
It looks like here we use a tuple of lists to represent a list of candidates, and for `CandidateSelector.__call__` we use an iterable of `Dict[str,str]`. To avoid inconsistency and to make these types a bit better understandable, can we create a specific type for one Candidate, represented by its ID + description?

Reply:
Done in https://github.com/explosion/spacy-llm/pull/241/commits/d6e347b1325a2e14c9dc69de3ddb6b642445d12a.